tests/regression/test_regression.py

```diff
@@ -0,0 +1,134 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from pathlib import Path
+
+import pytest
+from otx.cli.train import otx_train
+
+import mlflow
+
+
+@dataclass
+class ModelTestCase:
+    task: str
+    name: str
+
+
+@dataclass
+class DatasetTestCase:
+    name: str
+    data_root: Path
+    data_format: str
+    num_classes: int
+    extra_overrides: dict
+
+
+@dataclass
+class RegressionTestCase:
+    model: ModelTestCase
+    dataset: DatasetTestCase
+    output_dir: Path
+
+
+class BaseTest:
+    def _test_regression(
+        self,
+        model_test_case: ModelTestCase,
+        dataset_test_case: DatasetTestCase,
+        fxt_dataset_root_dir: Path,
+        fxt_tags: dict,
+        fxt_num_repeat: int,
+        tmpdir: pytest.TempdirFactory,
+    ) -> None:
+        for seed in range(fxt_num_repeat):
+            test_case = RegressionTestCase(
+                model=model_test_case,
+                dataset=dataset_test_case,
+                output_dir=Path(tmpdir) / str(seed),
+            )
+
+            run_name = f"{test_case.model.task}/{test_case.model.name}/{test_case.dataset.name}/{seed}"
+            tags = {
+                "task": test_case.model.task,
+                "model": test_case.model.name,
+                "dataset": test_case.dataset.name,
+                "seed": str(seed),
+                **fxt_tags,
+            }
+            data_root = (
+                fxt_dataset_root_dir
+                / test_case.model.task
+                / test_case.dataset.data_root
+            )
+            with mlflow.start_run(tags=tags, run_name=run_name):
+                overrides = [
+                    f"+recipe={test_case.model.task}/{test_case.model.name}",
+                    f"model.otx_model.config.head.num_classes={test_case.dataset.num_classes}",
+                    f"data.data_root={data_root}",
+                    f"data.data_format={test_case.dataset.data_format}",
+                    f"base.output_dir={test_case.output_dir}",
+                    f"seed={seed}",
+                    "test=true",
+                ] + [
+                    f"{key}={value}"
+                    for key, value in test_case.dataset.extra_overrides.items()
+                ]
+                _, metrics = otx_train(overrides)
+
+                # Submit metrics to MLFlow Tracker server
+                mlflow.log_metrics(metrics)
+
+
+class TestMultiClassCls(BaseTest):
```

Comment:
@kprokofi, @eugene123tw, @sungmanc, @jaegukhyun 

I'd like to request you guys to read this carefully and horizontally scale this test code for tasks you already took after this PR.